### PR TITLE
using the new test organisation from prettier

### PR DESCRIPTION
**Pull request by Janther**
the organisation makes a lot of sense and gives us a standard.

### CODE REVIEW

**Comment by Janther** [comment]
if anyone had an idea of how to move the contents of `tests_config` to `tests/config` that would be great as I had no success.

File: .prettierignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
 coverage/**/*.js
-tests/**/*.sol
-tests_config/**/*.js
```
Comment (mattiaerre) [inline .prettierignore:3]: should we add `tests/config` to this file too?

File: tests/config/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "singleQuote": false,
+    "trailingComma": "es5"
+  }
+
```
Comment (mattiaerre) [inline tests/config/.prettierrc:5]: can you add a new line here? thanks

File: tests/config/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "singleQuote": false,
```
Comment (mattiaerre) [inline tests/config/.prettierrc:2]: double quotes now? really?

File: tests/config/format-test.js [hunk 4]
```diff
@@ -1,30 +1,32 @@
+"use strict";
```
Comment (mattiaerre) [inline tests/config/format-test.js:1]: why did we add this? is this not always the case?

File: tests/config/.prettierrc [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+    "trailingComma": "es5"
```
Comment (mattiaerre) [inline tests/config/.prettierrc:3]: and trailing comma too? oh noes 😱

File: tests/config/require-prettier.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+"use strict";
+
+const prettier = require("prettier");
+
+module.exports = prettier;
```
Comment (mattiaerre) [inline tests/config/require-prettier.js:5]: what is this file for?

File: tests/unit/binary-operator-printers/index.test.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-const binaryOperatorPrinters = require('../../src/binary-operator-printers');
+const binaryOperatorPrinters = require('../../../src/binary-operator-printers');
```
Comment (mattiaerre) [inline tests/unit/binary-operator-printers/index.test.js:1]: single quotes here?

**Review by mattiaerre — COMMENTED** [review]
added some comments, I will add more later I think

**Comment by mattiaerre** [comment]
@Janther can you add a link to where this "inspiration" is coming from? I think I am missing some context here 😊

File: .prettierignore (hunk 1 above)
Comment (Janther) [reply .prettierignore:3]: On this folder I added the `.prettierrc` with the default that prettier uses so we still benefit from using prettier on the small changes that we make here for us while keeping their code untouched.

Comment (Janther) [reply .prettierignore:3]: eslint on the other hand should ignore this folder

File: tests/config/.prettierrc (hunk 3 above)
Comment (Janther) [reply tests/config/.prettierrc:2]: this is just for the files we took from the prettier repo using prettier's defaults

File: tests/config/format-test.js (hunk 4 above)
Comment (Janther) [reply tests/config/format-test.js:1]: I want to keep these files as close as possible to prettier's so it's easier to update them following their new practices

File: tests/config/.prettierrc (hunk 5 above)
Comment (Janther) [reply tests/config/.prettierrc:3]: same as the other comment

File: tests/unit/binary-operator-printers/index.test.js (hunk 7 above)
Comment (Janther) [reply tests/unit/binary-operator-printers/index.test.js:1]: this is our tests with our `.prettierrc` the default config is only for `tests/config`

File: tests/config/require-prettier.js (hunk 6 above)
Comment (Janther) [reply tests/config/require-prettier.js:5]: the tests can run using `prettier` or `prettier/standalone` as the engine.

**Comment by Janther** [comment]
> @Janther can you add a link to where this "inspiration" is coming from? I think I am missing some context here 😊

there are a lot of things that we decide on how to do regarding the formatting of code. But we lack some vision in 2 aspects:

  * Comments
  * Testing

That is why we are using `prettier`s code for these and as long as we do this, I'd like to keep a close look at what they are doing in these aspects and not be stale.

**Review by fvictorio — APPROVED** [review]

**Pull request merged**